edgelet/provisioning/src/provisioning.rs

```diff
@@ -116,19 +126,53 @@ impl From<&str> for ReprovisioningStatus {
             "deviceDataReset" => ReprovisioningStatus::DeviceDataReset,
             "initialAssignment" => ReprovisioningStatus::InitialAssignment,
             _ => {
-                debug!("DPS provisioning result substatus {}", s);
+                debug!("Provisioning result substatus {}", s);
                 ReprovisioningStatus::InitialAssignment
             }
         }
     }
 }
 
+impl Display for ReprovisioningStatus {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
 impl Default for ReprovisioningStatus {
     fn default() -> Self {
         ReprovisioningStatus::InitialAssignment
     }
 }
 
+impl From<&str> for ProvisioningStatus {
+    fn from(s: &str) -> ProvisioningStatus {
+        match s {
+            "assigned" => ProvisioningStatus::Assigned,
+            "assigning" => ProvisioningStatus::Assigning,
+            "disabled" => ProvisioningStatus::Disabled,
+            "failed" => ProvisioningStatus::Failed,
+            "unassigned" => ProvisioningStatus::Unassigned,
+            _ => {
+                debug!("Provisioning result status {}", s);
+                ProvisioningStatus::Assigned
```

Comment:
Is `Assigned` a safe default? I feel we should fail instead.

Reply:
Changed it to return an error by default